src/lib/fastDownward/missingSkillDetection.ts

```diff
@@ -157,7 +158,7 @@ export function skillAnalysis<LU extends LearningUnit>(
                 break;
             }
         }
-        goalString.split(",").filter(skill => skill.length > 0).forEach(sk => {
+        goalsArray.filter(skill => skill.length > 0).forEach(sk => {

```

Comment:
possible unnecessary filter.

in your case: if i assume if `skill.length` is zero, the `!openListMap.has(skill)` wont be true. If this is the case, then you can remove the filter here.

Reply:
Solved, remove the filter (not needed).